client/state/reader/follows/reducer.js

```diff
@@ -96,23 +96,32 @@ export const items = createReducer(
 			};
 		},
 		[ READER_FOLLOW ]: ( state, action ) => {
-			const urlKey = prepareComparableUrl( action.payload.feedUrl );
+			let urlKey = prepareComparableUrl( action.payload.feedUrl );
 			const newValues = { is_following: true };
 
-			// Reset follow error state
-			if ( state[ urlKey ] && state[ urlKey ].error ) {
+			const actualFeedUrl = get(
+				action.payload,
+				[ 'follow', 'feed_URL' ],
+				action.payload.feedUrl
+			);
+
+			const newState = { ...state };
+			if ( actualFeedUrl !== action.payload.feedUrl ) {
```

Comment:
add a single comment here saying:
`the case where a user follows by url and the actual feed_URL is different from what the user entered`